components/places/ios/Places/Places.swift

```diff
@@ -0,0 +1,418 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import os.log
+
+internal typealias ApiHandle = UInt64
+internal typealias ConnHandle = UInt64
+
+/**
+ * This is something like a places connection manager. It primarially exists to
+ * ensure that only a single write connection is active at once.
+ *
+ * If it helps, you can think of this as something like a connection pool
+ * (although it does not actually perform any pooling).
+ */
+public class PlacesApi {
+    private let handle: ApiHandle
+    private let writeConn: PlacesWriteConn
+    fileprivate let queue = DispatchQueue(label: "com.mozilla.places.api")
+
+    /**
+     * Initialize a PlacesApi
+     *
+     * - Parameter path: an absolute path to a file that will be used for the internal database.
+     *
+     * - Parameter encryptionKey: an optional key used for encrypting/decrypting data stored
+     *                            in the internal database. If omitted, data will be stored
+     *                            as plaintext.
+     *
+     * - Throws: `PlacesError` if initializing the database failed.
+     */
+    public init(path: String, encryptionKey: String? = nil) throws {
```

Comment:
I love the idea of making this optionally encrypted! 👍

Reply:
It can be a performance footgun, so I don't love it, but it's an option!